cmd/policy-assistant/pkg/matcher/traffic.go

```diff
@@ -71,10 +75,80 @@ func (p *TrafficPeer) IsExternal() bool {
 	return p.Internal == nil
 }
 
+func (p *TrafficPeer) Translate() TrafficPeer {
+	var podsNetworking []*PodNetworking
+	var podLabels map[string]string
+	var namespaceLabels map[string]string
+	var workloadOwner string
+	workloadOwnerExists := false
+	workloadMetadata := strings.Split(strings.ToLower(p.Internal.Workload), "/")
+	if (workloadMetadata[0] == "" || workloadMetadata[1] == "" || workloadMetadata[2] == "") || len(workloadMetadata) != 3 || (workloadMetadata[1] != "daemonset" && workloadMetadata[1] != "statefulset" && workloadMetadata[1] != "replicaset" && workloadMetadata[1] != "deployment" && workloadMetadata[1] != "pod") {
```

Comment:
nit: think this code could crash if `p.Internal.Workload = "abc"` for instance

Could move `len(workloadMetadata) != 3` first to make the boolean short-circuit

Reply:
Thanks, yes, true. I fixed it.